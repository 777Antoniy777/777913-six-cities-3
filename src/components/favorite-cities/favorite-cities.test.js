import React from 'react';
import renderer from "react-test-renderer";
import FavoriteCities from "./favorite-cities";

// set mocha data
const favoriteOffers = [
  {
    id: 1,
    city: {
      id: 1,
      name: `city`,
      coords: [1, 1],
    },
    title: `title 1`,
    premium: false,
    src: `img/image1`,
    photos: [`img/image1`],
    price: 999999,
    description: `test`,
    type: `type`,
    rating: 9999,
    bedroomAmount: 30,
    guestsAmount: 50,
    items: [`item`],
    host: {
      avatar: `img/avatar-1.jpg`,
      name: `name`,
      status: false,
    },
    reviews: [
      {
        id: 1,
        body: `text`,
        rating: 5,
        name: `name`,
        date: `date`,
      },
    ],
    coord: [1, 1],
  },
];
const favoriteCities = [
  `Moscow`,
  `Omsk`,
];
const history = {};
const location = {
  pathname: `/pathname`,
};

const getCurrentOffer = () => {};

it(`render FavoriteCities`, () => {
  const tree = renderer.create(
      <FavoriteCities
        favoriteOffers={favoriteOffers}
        favoriteCities={favoriteCities}
        history={history}
        location={location}
        getCurrentOffer={getCurrentOffer}
      />)
      .toJSON();

  expect(tree).toMatchSnapshot();
});
