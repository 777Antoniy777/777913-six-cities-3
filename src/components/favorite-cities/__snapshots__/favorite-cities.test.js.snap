// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render FavoriteCities 1`] = `
<ul
  className="favorites__list"
>
  <li
    className="favorites__locations-items"
  >
    <div
      className="favorites__locations locations locations--current"
    >
      <div
        className="locations__item"
      >
        <a
          className="locations__item-link"
          href="#"
        >
          <span>
            Moscow
          </span>
        </a>
      </div>
    </div>
    <div
      className="favorites__places"
    />
  </li>
  <li
    className="favorites__locations-items"
  >
    <div
      className="favorites__locations locations locations--current"
    >
      <div
        className="locations__item"
      >
        <a
          className="locations__item-link"
          href="#"
        >
          <span>
            Omsk
          </span>
        </a>
      </div>
    </div>
    <div
      className="favorites__places"
    />
  </li>
</ul>
`;
